### ISSUE TITLE
Remove p2p code from Synapse

### ISSUE THREAD

**Issue by cafca** [cafca/souma#64]

**Comment by cafca** [comment]
@cartisan @mrausch 

I noticed an interesting thing while working on this. The p2p code allowed discovering content just because it was cached on a machine in my network, so not directly because of a personal connection. 

Currently, the Myelin server allows Vesicles to be sent from 1 to n Personas. In the p2p code the same mechanism existed, with the difference that before this could happen an online Souma hosting the other Persona's content would have to be found using the Glia service - then the Vesicle could be sent directly to that Souma. While connecting to the new remote Souma, an index of the respective Souma's cached contents was exchanged, which of course included content from sources my Souma did not know about before. 

I have started implementing this in a server-based model, but now I think it's really best if we focus on the basics first. Having direct communication between Soumas is a nice feature that we can maybe pick up again when come back to implementing p2p. I especially like the idea of partially uncoupling data distribution from actual communication paths. Some advantages of that could be
- discovery of anonymous content
- compensating for mismatch between a user's hardware resources and his communication amount
- enabling plausible deniability
